### PR TITLE
antlr4-maven-plugin: recompile grammars when imports have changed

### PR DESCRIPTION
**Pull request by dhalperi**
The antlr4-maven-plugin already has dependency analysis that's supposed to do this, but it's got some bugs. I submitted several bug fixes upstream in https://github.com/antlr/antlr4/pull/1945.

This PR checks in a hacked version of the antlr4-maven-plugin with improved dependency analysis, as a stopgap until the fixes are accepted upstream.

### CODE REVIEW

**Comment by dhalperi** [comment]
This is for #195

**Comment by dhalperi** [comment]
Tho clearly, currently broken. Investigating.

File: projects/batfish/pom.xml [hunk 1]
```diff
@@ -53,6 +53,7 @@
             <configuration>
               <includes>**/CiscoLexer.g4,**/CiscoParser.g4</includes>
               <libDirectory>${basedir}/src/main/antlr4/org/batfish/grammar/cisco</libDirectory>
+              <statusDirectory>${project.build.directory}/maven-status/antlr4-cisco</statusDirectory>
```
Comment (arifogel) [inline projects/batfish/pom.xml]: Did you only need to do his for grammars with imports?

Comment (dhalperi) [inline projects/batfish/pom.xml]: yes

Comment (arifogel) [inline projects/batfish/pom.xml]: um tests are failing

Comment (dhalperi) [inline projects/batfish/pom.xml]: Yeah something about how I'm installing the plugin is breaking vs working when installing dependencies.

**Pull request merged**